File: springboot2.x-learning/springboot-data-mongodb/src/main/java/com/rtxtitanv/projections/NameOnly.java
package com.rtxtitanv.projections;

/**
 * @author rtxtitanv
 * @version 1.0.0
 * @name com.rtxtitanv.projections.NameOnly
 * @description 将查询结果限制为只有名称属性的projections
 * @date 2021/5/27 14:34
 */
public interface NameOnly {
    String getUsername();

    String getRealname();
}

File: springboot2.x-learning/springboot-data-jpa-kotlin/src/main/kotlin/com/rtxtitanv/projections/NameOnly.kt
package com.rtxtitanv.projections

/**
 * @name com.rtxtitanv.projections.NameOnly
 * @description 只查询name的projections
 * @author rtxtitanv
 * @date 2020/1/17 17:07
 * @version 1.0.0
 */
interface NameOnly {
    fun getUserName(): String? //? 这里表示声明返回值为Nullable，即返回值允许为null
    fun getNickName(): String?
}